### PR TITLE
Proposed 1.9.4

### PR DESCRIPTION
**Pull request by nbougalis**
This release primarily introduces an amendment to address the potential denial-of-service attack against NFT issuers, by removing the `tfTrustLine` flag which is required for the attack.

If merged, this PR will close #4301, close #4298, and close #4283.

### CODE REVIEW

**Comment by Cryptodamus113** [comment]
🦉

**Review by ximinez — CHANGES_REQUESTED** [review]

File: RELEASENOTES.md [hunk 1]
```diff
@@ -7,6 +7,50 @@ This document contains the release notes for `rippled`, the reference server imp
  
 Have new ideas? Need help with setting up your node? Come visit us [here](https://github.com/xrplf/rippled/issues/new/choose)
 
+# Introducing XRP Ledger version 1.9.4
+
+Version 1.9.4 of `rippled`, the reference implementation of the XRP Ledger protocol is now available. This releases introduces an amendment that removes the ability for an NFT issuer to indicate that trustlines should be automatically created for royalty payments from secondary sales of NFTs, in response to a bug report that indicated how this functionality could be abused to mount a ddenial of service attack against the issuer.
```
Comment (rippleitinnz) [inline RELEASENOTES.md:12]: Typo 'This releases...' should be 'This release...'

Comment (rippleitinnz) [inline RELEASENOTES.md:12]: Type 'ddenial of service' should conform with other usage - as in 16 - 'denial-of-service'

Comment (intelliot) [inline RELEASENOTES.md:12]:
```suggestion
Version 1.9.4 of `rippled`, the reference implementation of the XRP Ledger protocol is now available. This releases introduces an amendment that removes the ability for an NFT issuer to indicate that trust lines should be automatically created for royalty payments from secondary sales of NFTs, in response to a bug report that indicated how this functionality could be abused to mount a denial of service attack against the issuer.
```

File: RELEASENOTES.md [hunk 2]
```diff
@@ -7,6 +7,50 @@ This document contains the release notes for `rippled`, the reference server imp
...
+
+## Action Required
+
+This release introduces a new amendment to the XRP Ledger protocol, **`fixRemoveNFTokenAutoTrustLine`** to mitigate a potential denial-of-service attack against NFT issuers that minted NFTs and allowed secondary trading of those NFTs to create trustlines for any asset.
```
Comment (intelliot) [inline RELEASENOTES.md:16]:
```suggestion
This release introduces a new amendment to the XRP Ledger protocol, **`fixRemoveNFTokenAutoTrustLine`** to mitigate a potential denial-of-service attack against NFT issuers that minted NFTs and allowed secondary trading of those NFTs to create trust lines for any asset.
```

**Review by intelliot — COMMENTED** [review]
In prose, "trust line" is two words, which is why the L in `TrustLine` is capitalized (in `fixRemoveNFTokenAutoTrustLine`, `tfTrustLine`, etc.)

File: RELEASENOTES.md [hunk 3]
```diff
@@ -7,6 +7,50 @@ This document contains the release notes for `rippled`, the reference server imp
...
+
+## Install / Upgrade
+
+On supported platforms, see the [instructions on installing or updating `rippled`](https://xrpl.org/install-rippled.html).
+
+## Changelog
+
+## Contributions
+
+This primary change in this release is the following bug fix:
```
Comment (rippleitinnz) [inline RELEASENOTES.md:33]: Grammar 'This primary change in this ...' should be 'The primary change in this...'

File: RELEASENOTES.md [hunk 4]
```diff
@@ -7,6 +7,50 @@ This document contains the release notes for `rippled`, the reference server imp
...
+
+- **Introduce fixRemoveNFTokenAutoTrustLine amendment**: Introduces the `fixRemoveNFTokenAutoTrustLine` amendment, which disables the `tfTrustLine` flag, which a malicious attacker could exploit to mount denial of service attacks against NFT issuers that specified the flag on their NFTs. ([#4301](https://github.com/XRPLF/rippled/4301))
+
+
+### GitHub
+
+The public source code repository for `rippled` is hosted on GitHub at <https://github.com/XRPLF/rippled>.
+
+We welcome contributions, big and small, and invite everyone to join the community of XRP Ledger developers and help us build the Internet of Value.
```
Comment (rippleitinnz) [inline RELEASENOTES.md:42]: Sentence structure: 'We welcome contributions, big and small, and...'

Consider 'We welcome all contributions and...'

**Review by rippleitinnz — CHANGES_REQUESTED** [review]

**Comment by Cryptodamus113** [comment]
If we’re in a sim can someone here code me rich again?

File: RELEASENOTES.md (hunk 1 above)
Comment (nbougalis) [reply RELEASENOTES.md:12]: I've been calling it a trustline for a decade, and a lot of the documentation does as well, but I'm ok with changing this.

**Review by rippleitinnz — APPROVED** [review]

**Comment by myalcin81** [comment]
this amendment will be voted yes as default but it requires the other last amendment should pass to be enabled.
so is this the way good to go?

ps. my comment is just about how it works as well as a reminder because we had some issues recently and made lots of fixes for that and No one wants to introduce a new deprecated amendment again.

**Review by whywefight — APPROVED** [review]

**Review by ximinez — CHANGES_REQUESTED** [review]

**Review by nixer89 — APPROVED** [review]
lgtm

**Comment by nixer89** [comment]
> this amendment will be voted yes as default but it requires the other last amendment should pass to be enabled. so is this the way good to go?
> 
> ps. my comment is just about how it works as well as a reminder because we had some issues recently and made lots of fixes for that and No one wants to introduce a new deprecated amendment again.

yes, this fix amendment has the default vote set to "Yes". That is because it won't have any effect until the amendment `NonFungibleTokensV1_1` is enabled.

**Comment by rippleitinnz** [comment]
> > this amendment will be voted yes as default but it requires the other last amendment should pass to be enabled. so is this the way good to go?
> 
> > 
> 
> > ps. my comment is just about how it works as well as a reminder because we had some issues recently and made lots of fixes for that and No one wants to introduce a new deprecated amendment again.
> 
> 
> 
> yes, this fix amendment has the default vote set to "Yes". That is because it won't have any effect until the amendment `NonFungibleTokensV1_1` is enabled.

It will have an effect in amendment blocking all those on versions earlier than 1.9.4 if it passes and becomes 'able' to be enabled.

**Comment by manojsdoshi** [comment]
Suspended. Using https://github.com/XRPLF/rippled/pull/4307 for releasing 1.9.4

**Pull request closed without merging**